Helper/ApiClient.php

```diff
@@ -29,56 +29,99 @@ public function __construct(\Magento\Store\Model\StoreManagerInterface $storeMan
+                        $productID = $productIDs[0];
+                    }
+                    if ($productCollection->create()->count()) {
+                        $productCollection->create()->clear();
+                        $_product = $item->getProduct();
+                    }
+                    $_products = $productCollection->create()->addAttributeToSelect('*')->addStoreFilter()->addFieldToFilter('entity_id', ['in' => $productID]);
+                    foreach ($_products as $product) { // This is needed as we can't use object as array in collection way.
+                        $_product = $product;
+
+                        break 1;
+                    }
+                }
+
+                $productName = $_product->getName();
+                $productUrl = $_product->getProductUrl();
+                $imageUrl = $store->getBaseUrl(\Magento\Framework\UrlInterface::URL_TYPE_MEDIA) . 'catalog/product' . $_product->getImage();
+                $sku = $_product->getSku();
+                $upc = $_product->getUpc();
+                $isbn = $_product->getIsbn();
+                $mpn = $_product->getMpn();
+                $brand = $_product->getBrand();
+
+
+                $product_data['name'] = $productName;
+                $product_data['url'] = '';
+                $product_data['image'] = '';
+                $product_data['url'] = $productUrl;
+                $product_data['image'] = $imageUrl;
+                if ($upc) {
+                    $specs_data['upc'] = $upc;
+                }
+                if ($isbn) {
+                    $specs_data['isbn'] = $isbn;
+                }
+                if ($brand) {
+                    $specs_data['brand'] = $brand;
+                }
+                if ($mpn) {
+                    $specs_data['mpn'] = $mpn;
+                }
+                if ($sku) {
+                    $specs_data['external_sku'] = $sku;
+                }
+                if (!empty($specs_data)) {
+                    $product_data['specs'] = $specs_data;
+                }
+            } catch (\Exception $e) {
+                $this->_logger->addDebug('ApiClient prepareProductsData Exception' . json_encode($e));
+            }
+            $rawdescription = str_replace(array('\'', '"'), '', $_product->getDescription());
+            $description = $this->_escaper->escapeHtml(strip_tags($rawdescription));
+            $product_data['description'] = $description;
+            if ($item->getData('product_type') == 'grouped') {
+                if (!isset($productPrice[$productID])) {
+                    $productPrice[$productID] = 0.0000;
+                }
+                $productPrice[$productID] += $item->getData('row_total_incl_tax');
+                $product_data['price'] = $productPrice[$productID];
+            } else {
+                if (!isset($productPrice[$productID])) {
```

Comment:
The else statement is identical to the if statement, in both cases the same code executed.